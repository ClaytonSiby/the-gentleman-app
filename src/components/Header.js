import React from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import logo from '../assets/images/logo.PNG';
import styles from '../assets/scss/header.module.scss';

const Header = () => {
  const userToken = localStorage.getItem('userToken');
  return (
    <Navbar collapseOnSelect expand="sm" className={`${styles.theNavbar}`}>
      <Navbar.Brand href="/"><img className="logo" src={logo} alt="Car Rentals" /></Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto" />
        <Nav className="d-flex justify-content-end">
          {
        userToken === undefined || userToken === 'undefined' || userToken === '' || userToken === null
          ? (
            <>
              <a className="nav-link white" href="/">Home</a>
              <a className="nav-link white" href="/login">LOGIN</a>
              <a className="nav-link" href="/signup">
                SIGNUP
              </a>
            </>
          )
          : (
            <>
              <a className="nav-link white" href="/">Home</a>
              <a className="nav-link white" href="/suits">Suits</a>
              <a className="nav-link white" href="/appointments">Appointments</a>
              <a className="nav-link white" href="/logout">
                LOGOUT
              </a>
            </>
          )
      }
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

export default Header;
