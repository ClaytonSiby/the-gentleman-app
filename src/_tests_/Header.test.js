import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Header', () => {
  test('renders Header component', () => {
    render(<Header />);
  });

  test('renders consistently to the DOM', () => {
    const test = renderer.create(<Header />).toJSON();

    expect(test).toMatchSnapshot();
  })
});
