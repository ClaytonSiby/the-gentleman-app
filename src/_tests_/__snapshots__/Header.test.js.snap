// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders consistently to the DOM 1`] = `
<nav
  className="theNavbar navbar navbar-expand-sm navbar-light"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <img
      alt="Car Rentals"
      className="logo"
      src="logo.PNG"
    />
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="d-flex justify-content-end navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link white"
        href="/"
      >
        Home
      </a>
      <a
        className="nav-link white"
        href="/suits"
      >
        Suits
      </a>
      <a
        className="nav-link white"
        href="/appointments"
      >
        Appointments
      </a>
      <a
        className="nav-link white"
        href="/logout"
      >
        LOGOUT
      </a>
    </div>
  </div>
</nav>
`;
